tket/src/PauliGraphRefactor/PauliGraphConverters.cpp

```diff
@@ -306,144 +413,261 @@ pg::PauliGraph circuit_to_pauli_graph3(
   return res;
 }
 
-std::pair<Op_ptr, unit_vector_t> pgop_to_inner_command(
-    const PGOp_ptr& pgop, const qubit_vector_t& qubits) {
+struct IndividualSynth {
+  std::list<ChoiMixTableau::row_tensor_t> rows;
+  Op_ptr op;
+  unit_vector_t args;
+};
+
+enum class MultiplexorCase { Rotation, U2, TensoredU2, General };
+
+IndividualSynth synth_pgop(const PGOp_ptr& pgop) {
   switch (pgop->get_type()) {
     case PGOpType::Rotation: {
-      const PGRotation& rot = dynamic_cast<const PGRotation&>(*pgop);
-      return {get_op_ptr(OpType::Rz, rot.get_angle()), {qubits.front()}};
+      PGRotation& rot = dynamic_cast<PGRotation&>(*pgop);
+      return IndividualSynth{
+          {{rot.port(0), SpPauliStabiliser(Qubit(0), Pauli::Z)}},
```

Comment:
In the new PauliGraph, there are two kinds of dependencies tracked in different graphs:
- The classical data dependencies are described by a graph with one vertex per PGOp. The kinds of rewrites that are suited to the PauliGraph structure are unlikely to make significant changes to the classical data so there is no need to track lots of information in the graph as it doesn't need to be updated a lot.
- The quantum non-commutation dependencies. For the PGOps with multiple Pauli strings, ALL of them need to commute with others in order for two PGOps to commute. Rewrites on the graph will change individual Pauli strings on each PGOp, so recalculating the dependencies is much easier if we track the commutation between every Pauli string in the graph - we store it densely like an adjacency matrix so that updates (pretty much always multiplying two Pauli strings together) are simple row and column sums. This effectively makes it a graph with one vertex for each Pauli string in the graph. Each PGOp is now represented by a cluster of vertices, in much the same way that a port graph can be viewed as a graph between the ports with vertices now being a cluster of ports. This similarity gave rise to the term `port` for indexed Pauli strings within a PGOp